app/concerns/permission_management.rb

```diff
@@ -74,52 +97,96 @@ def assemble_permissions_for_updating(full_permissions:, type:, group_id:)
     assembled_permissions
   end
 
-  def sort_permissions_to_update(assembled_all_permissions:, permissions_params:)
+  def sort_permissions_to_update(assembled_all_permissions:, permissions_params:, type:)
     permissions_params ||= {}
 
     targets_to_add_group = []
     targets_to_update_perms = []
     targets_to_remove_group = []
     targets_to_delete = []
     targets_to_skip = []
+    target_revision_ids = {}
+    targets_to_fail = []
+    targets_to_create = []
+
+    current_revisions = get_revisions_for_edit(type: type)
 
     assembled_all_permissions.each do |target, acl_info|
       if acl_info['matched_to_group'] # group is currently in the target acl
         if acl_info['granted_permissions'] == permissions_params[target]
           # permissions are the same, do nothing
           targets_to_skip << target
         elsif permissions_params[target] == [] && acl_info['num_groups'] > 1
-          # removing permissions to the target, need to remove the group from the acl
-          targets_to_remove_group << target
+          # The acl_info is information obtained AFTER submission of the form
+          # The revision_id from the parameters is obtained BEFORE submission of the form
+          if params["#{target}_revision_id"].present?
```

Comment:
can this be a part of permissions_params?

Reply:
With the revision ids provided by the assemble function, the overwriting checks can and are being done in the controller.  As a consequence, there is no longer a need to pass revision ids in or use them from the params.